benchopt/util.py

```diff
@@ -103,42 +92,50 @@ def _run_bash_in_env(script, env_name=None, raise_on_error=None,
         Exit code of the script
     """
     if env_name is not None:
-        env_dir = f"{VENV_DIR}/{env_name}"
-
-        script = f"source {env_dir}/bin/activate\n{script}"
+        script = (f". activate {env_name} && {script}")
```

Comment:
Don't use the `&&` syntax here. The script can be multi-lines so it won't prevent for running the next lines.
To protect this, we set L.48 in `_run_in_bash` the `set -e` option that kills the script with the first error and return the exit code.